### PR TITLE
tests: make sure OpenSL ES rejects bad formats

### PR DESCRIPTION
**Pull request by philburk**

### CODE REVIEW

File: tests/testStreamOpen.cpp [hunk 1]
```diff
@@ -85,14 +85,39 @@ TEST_F(StreamOpen, OutputForOpenSLESPerformanceModeShouldBeNone){
...
 	ASSERT_EQ((int)mStream->getPerformanceMode(), (int)PerformanceMode::None);
     closeStream();
 }
+
+TEST_F(StreamOpen, ForOpenSlesIllegalFormatRejectedOutput) {
+    mBuilder.setAudioApi(AudioApi::OpenSLES);
+    mBuilder.setPerformanceMode(PerformanceMode::LowLatency);
+    mBuilder.setFormat(static_cast<AudioFormat>(666));
+    Result r = mBuilder.openStream(&mStream);
+    EXPECT_NE(r, Result::OK) << "Should not open stream " << convertToText(r);
```
Comment (dturner) [inline tests/testStreamOpen.cpp]: Wouldn't it be better to check for `oboe::Result::ErrorInvalidFormat`?

File: tests/testStreamOpen.cpp [hunk 2]
```diff
@@ -85,14 +85,39 @@ TEST_F(StreamOpen, OutputForOpenSLESPerformanceModeShouldBeNone){
...
+    }
+}
+
+TEST_F(StreamOpen, ForOpenSlesIllegalFormatRejectedInput) {
+    mBuilder.setAudioApi(AudioApi::OpenSLES);
+    mBuilder.setPerformanceMode(PerformanceMode::LowLatency);
+    mBuilder.setDirection(Direction::Input);
+    mBuilder.setFormat(static_cast<AudioFormat>(666));
+    Result r = mBuilder.openStream(&mStream);
+    EXPECT_NE(r, Result::OK) << "Should not open stream " << convertToText(r);
```
Comment (dturner) [inline tests/testStreamOpen.cpp]: Wouldn't it be better to check for `oboe::Result::ErrorInvalidFormat`?

File: tests/testStreamOpen.cpp (hunk 1 above)
Comment (philburk) [reply tests/testStreamOpen.cpp:147]: It is currently returning ErrorInternal.  

I think we should add some more general parameter checking to AudioStreamBuilder. That will get called in the open() and return a more suitable error message.

File: tests/testStreamOpen.cpp (hunk 2 above)
Comment (philburk) [reply tests/testStreamOpen.cpp:159]: ditto

**Comment by philburk** [comment]
The rebase and merge is not working because this is too old.
So I will resubmit this on a different branch.

**Pull request closed without merging**